### PR TITLE
Volume layer override

### PR DESCRIPTION
**Pull request by gaborkb**
### Purpose of this PR

https://jira.unity3d.com/browse/CMCL-284

CM VolumeSettings extension builds a hidden volume under the hood which arbitrarily chooses a layer from the camera's culling mask, so that it's visible.

The problem is that some users want to be able to choose which layer the volume attaches to. 

Description here: https://forum.unity.com/threads/hdrp-splitscreen-cinemachine-volume-settings-extension-issue.1043635/#post-6997481

Possible solution: an advanced setting allowing user to manually configure the layer.  By default it does what it currently does: chooses one arbitrarily

### Testing status

[Explanation of what’s tested, how tested and existing or new automation tests. Can include manual testing by self and/or QA. Specify test plans. Rarely acceptable to have no testing.]

- [ ] Added an automated test
- [ ] Passed all automated tests
- [x] Manually tested 

### Documentation status

[Overview of how documentation is affected by this change. If there is no effect on documentation, explain why. Otherwise, state which sections are changed and why.]

- [x] Updated [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
- [ ] Updated README (if applicable)
- [ ] Commented all public classes, properties, and methods
- [ ] Updated user documentation

### Samples status

- [ ] Re-packed Extras~/CinemachineExamples.unitypackage

### Technical risk

[Overall product level assessment of risk of change. Need technical risk & halo effect.]

### Comments to reviewers

[Info per person for what to focus on, or historical info to understand who have previously reviewed and coverage. Help them get context.]

### Package version

[Justification for updating either the patch, minor, or major version according to the [semantic versioning](https://semver.org/spec/v2.0.0.html) rules]

- [ ] Updated package version

### CODE REVIEW

**Comment by gaborkb** [comment]
I am not entirely sure about the use case. I tested only that I can set a layer that will make the volume invisible.

**Review by glabute — CHANGES_REQUESTED** [review]
Maybe we need to think about whether an advanced setting is the right solution.

File: Runtime/PostProcessing/CinemachineVolumeSettings.cs [hunk 1]
```diff
@@ -99,6 +99,18 @@ public enum FocusTrackingMode
...
+        /// Global override for the layer used by CinemachineVolumes.
+        /// </summary>
+        [Tooltip("Global override for the layer used by CinemachineVolumes. ")]
+        public static LayerMask s_LayerMaskOverride;
+
+        /// <summary>
+        /// If true, enables global layer override.
+        /// </summary>
+        [Tooltip("If true, enables global layer override. ")]
+        public static bool s_LayerMaskOverrideEnabled;
```
Comment (glabute) [inline Runtime/PostProcessing/CinemachineVolumeSettings.cs:112]: Why static?  Having it as a global setting does not respond to the needs.  If you read the forum post, the guy wants a per-camera setting.

File: Runtime/PostProcessing/CinemachineVolumeSettings.cs [hunk 2]
```diff
@@ -243,7 +257,9 @@ static void ApplyPostFX(CinemachineBrain brain)
         {
             CameraState state = brain.CurrentCameraState;
             int numBlendables = state.NumCustomBlendables;
-            var volumes = GetDynamicBrainVolumes(brain, numBlendables);
+            var currentVolume = 
+                brain.ActiveVirtualCamera.VirtualCameraGameObject.GetComponent<CinemachineVolumeSettings>();
+            var volumes = GetDynamicBrainVolumes(brain, numBlendables, currentVolume);
```
Comment (gaborkb) [inline Runtime/PostProcessing/CinemachineVolumeSettings.cs:262]: I dislike this part. We could discuss a better solution.

**Comment by gaborkb** [comment]
Marked draft until we come back to it.

**Comment by gaborkb** [comment]
See https://jira.unity3d.com/browse/CMCL-284 for more info.

**Pull request closed without merging**